File: src/web/magmac/api/collect/list/Lists.ts
export class Lists {
	public static of( ...elements : T[]) : List<T> {return new JVMList<>( new ArrayList<>( Arrays.asList( elements)));;}
	public static empty() : List<T> {return new JVMList<>( );;}
	public static repeat( element : T,  size : int) : List<T> { let copy : List<T>=Lists.empty( ); let i : var=0;if(true){ copy=copy.addLast( element);i++;;}return copy;;}
}

File: packages/compiler-java/src/main/java/magmac/api/collect/list/Lists.java
package magmac.api.collect.list;

import java.util.ArrayList;
import java.util.Arrays;

/**
 * Factory and helper methods for working with {@link List} values.
 */

public final class Lists {
    /**
     * Creates a list containing the given elements.
     */
    public static <T> List<T> of(T... elements) {
        return new JVMList<>(new ArrayList<>(Arrays.asList(elements)));
    }

    /**
     * Returns an empty list.
     */
    public static <T> List<T> empty() {
        return new JVMList<>();
    }

    /**
     * Creates a list repeating {@code element} {@code size} times.
     */
    public static <T> List<T> repeat(T element, int size) {
        List<T> copy = Lists.empty();
        var i = 0;
        while (i < size) {
            copy = copy.addLast(element);
            i++;
        }
        return copy;
    }
}
